addons/membership/membership.py

```diff
@@ -136,6 +137,11 @@ def _state(self, cr, uid, ids, name, args, context=None):
             res[line.id] = state
         return res
 
+    def _set_state(self, cr, uid, ids, field_name, field_value, args, context=None):
+        cr.execute('UPDATE membership_membership_line '\
+                       'SET state=%s '\
+                       'WHERE id IN %s', (field_value, tuple([ids]),))
+        return True
```

Comment:
I don't understand why do you want to force the write on a stored-function field via an inverse function on this field. If you want to update the `state` field when an invoice is canceld, why don't you adapt the recomputation triggers for the `state` field accordingly?